src/docpage.js

```diff
@@ -3,19 +3,201 @@ import React from 'react'
 export default class DocPage extends React.Component {
 
   componentWillMount() {
-    let docUrl = window.location.href.split('?')[0]
-    let dataUrl = `${docUrl}/json`
-    $.get(dataUrl, (doc) => {
-      this.setState({doc})
+    let docUrl = this.props.docUrl || window.location.href.split('?')[0]
+    let split = docUrl.split('/'),
+        docId = split.pop(),
+        baseUrl = split.join('/')
+    this.baseUrl = baseUrl
+
+    $.get(`${docUrl}/json`, (doc) => {
+      this.setState({doc: doc, loaded: true})
+    })
+  }
+
+  render() {
+    let state = this.state || {},
+        doc = state.doc || {},
```

Comment:
Please use a separate `let` statement for each variable.

(this message was brought to you by the _Unwritten Coding Style Guide_)